internal/common/sources_aspect.bzl

```diff
@@ -34,7 +34,8 @@ def _sources_aspect_impl(target, ctx):
   if hasattr(target, "typescript"):
     result = depset(transitive=[result, target.typescript.es5_sources])
   elif hasattr(target, "files"):
-    result = depset(transitive=[result, target.files])
+    result = depset([f for f in target.files if f.path.endswith(".js")],
```

Comment:
Intentionally outside of transitive?

Reply:
yes, because this is a list now that I've iterated and filtered it, not a depset